### PR TITLE
[#1903] feat(clients): add equals for `GravitinoMetalake`.

### PR DESCRIPTION
**Pull request by LindaSummer**
### What changes were proposed in this pull request?

Add `equals()` and unit tests for `GravitinoMetalake`.

### Why are the changes needed?

As issue mentioned, `GravitinoMetalake` class should have an equals method.

Fix: #1903

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test cases has been added.

### CODE REVIEW

**Comment by justinmclean** [comment]
There are some code formatting issues here. You'll need to run `./gradlew :clients:client-java:spotlessApply` to correct and commit the changes.

**Comment by LindaSummer** [comment]
> There are some code formatting issues here. You'll need to run `./gradlew :clients:client-java:spotlessApply` to correct and commit the changes.

Thanks very much for your suggestions!
I will fix the formatting issues with this command.

Best Regards,
Edward

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java [hunk 1]
```diff
@@ -843,6 +844,20 @@ public GravitinoMetalake build() {
...
+    if (this == o) {
+      return true;
+    }
+
+    if (!(o instanceof GravitinoMetalake)) {
+      return false;
+    }
+
+    GravitinoMetalake that = (GravitinoMetalake) o;
+    return super.equals(that) && Objects.equal(restClient, that.restClient);
```
Comment (jerryshao) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java:858]: I feel like that we don't need to compare the `restClient` object. If the same metalake is gotten from different clients, I think though `restClient` is different, the `GravitinoMetalake` itself should be the same.

Comment (LindaSummer) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java:858]: Hi @jerryshao ,

Thanks for your suggestions!
I will remove the `restClient` comparison.

Best Regards,
Edward

**Review by jerryshao — APPROVED** [review]

**Comment by jerryshao** [comment]
Thanks @LindaSummer for your contribution, merging to main and branch-0.6.

**Pull request merged**